lib/archethic.ex

```diff
@@ -116,10 +121,68 @@ defmodule Archethic do
     message = %StartMining{
       transaction: tx,
       welcome_node_public_key: get_welcome_node_public_key(tx_type, welcome_node_key),
-      validation_node_public_keys: Enum.map(validation_nodes, & &1.last_public_key)
+      validation_node_public_keys: Enum.map(validation_nodes, & &1.last_public_key),
+      network_chains_view_hash: NetworkView.get_chains_hash(),
+      p2p_view_hash: NetworkView.get_p2p_hash()
     }
 
-    P2P.broadcast_message(validation_nodes, message)
+    Task.Supervisor.async_stream_nolink(
+      Archethic.TaskSupervisor,
+      validation_nodes,
+      &P2P.send_message(&1, message),
+      ordered: false,
+      on_timeout: :kill_task,
+      timeout: Message.get_timeout(message) + 2000
+    )
+    |> Stream.filter(&match?({:ok, _}, &1))
+    |> Stream.map(fn {:ok, res} -> res end)
+    |> Enum.reduce(
+      %{
+        ok: 0,
+        network_chains_resync_needed: false,
+        p2p_resync_needed: false
+      },
+      fn
+        {:ok, %Ok{}}, acc ->
+          %{acc | ok: acc.ok + 1}
+
+        {:ok, %Error{reason: :network_chains_sync}}, acc ->
+          %{acc | network_chains_resync_needed: true}
+
+        {:ok, %Error{reason: :p2p_sync}}, acc ->
+          %{acc | p2p_resync_needed: true}
+
+        {:ok, %Error{reason: :both_sync}}, acc ->
+          %{
+            acc
+            | network_chains_resync_needed: true,
+              p2p_resync_needed: true
+          }
+
+        _, acc ->
+          acc
+      end
+    )
+    |> then(fn result ->
```

Comment:
I think you should use a private function; it will be better to image what the function & pipeline does without having to deep dive into the code.